ccdproc/extern/bitmask.py

```diff
@@ -0,0 +1,235 @@
+# External license! License can be found in "licenses/LICENSE_BITMASK.txt".
+
+"""
+A module that provides functions for manipulating bitmasks and data quality (DQ) arrays.
+
+:Authors: Mihai Cara (contact: help@stsci.edu)
+
+:License: `<http://www.stsci.edu/resources/software_hardware/pyraf/LICENSE>`_
+
+"""
+
+import numpy as np
+
+__version__ = '0.2.0'
+__vdate__ = '23-February-2017'
+__author__ = 'Mihai Cara'
+
+# Revision history:
+# 0.1.0 (29-March-2015) - initial release based on code from stsci.skypac
+# 0.1.1 (21-February-2017) - documentation typo fix
+# 0.2.0 (23-February-2017) - performance and stability improvements. Changed
+#       default output mask type from numpy.uint8 to numpy.bool_.
+
+
+def interpret_bits_value(val):
+    """
+    Converts input bits value from string to a single integer value or None.
+    If a comma- or '+'-separated set of values are provided, they are summed.
+
+    .. note::
+        In order to flip the bits of the final result (after summation),
+        for input of `str` type, prepend '~' to the input string. '~' must
+        be prepended to the *entire string* and not to each bit flag!
+
+    Parameters
+    ----------
+    val : int, str, None
+        An integer bit mask or flag, `None`, or a comma- or '+'-separated
+        string list of integer bit values. If `val` is a `str` and if
+        it is prepended with '~', then the output bit mask will have its
+        bits flipped (compared to simple sum of input val).
+
+    Returns
+    -------
+    bitmask : int or None
+        Returns and integer bit mask formed from the input bit value
+        or `None` if input `val` parameter is `None` or an empty string.
+        If input string value was prepended with '~', then returned
+        value will have its bits flipped (inverse mask).
+
+    Examples
+    --------
+        >>> from ccdproc.extern import bitmask
+        >>> "{0:016b}".format(0xFFFF & bitmask.interpret_bits_value(28) )
```

Comment:
@MSeifert04 I am working on redesigning stsci.tools.bitmask to add some new features (e.g., support for Python lists of flags) and improved terminology. In particular I do not like the use of "bit value" instead of flags. In my mind, *bit value* is either 0 or 1 and not ``2**n`` (which is a flag).